CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
+PROJECT(planck)
+
+INCLUDE(GNUInstallDirs)
+INCLUDE(FindPkgConfig)
+
+SET(VERSION_MAJOR 0)
+SET(VERSION_MINOR 1)
+SET(VERSION_PATCH 0)
+SET(VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
+
+# Common compile options
+ADD_COMPILE_OPTIONS(
+	-g -Os
+
+	# Turn on all warnings and treat all warnings as errors.
+	-Werror -Wall -Wextra
+
+	# Additional warnings
+	-Wunreachable-code -Wpointer-arith
+	-Wformat=2 -Wno-format-nonliteral
+
+	# Not warnings
+	-Wno-missing-field-initializers
+	-Wno-unused-parameter)
+
+ADD_DEFINITIONS(-DVERSION=\"${VERSION}\")
+
+INCLUDE_DIRECTORIES(
+	${CMAKE_BINARY_DIR}
+	${CMAKE_SOURCE_DIR}/src)
+
+# lex & yacc
+ADD_CUSTOM_COMMAND(
+	OUTPUT ${CMAKE_BINARY_DIR}/y.tab.c ${CMAKE_BINARY_DIR}/y.tab.h
+	COMMAND yacc -d ${CMAKE_SOURCE_DIR}/syntax/syntax.yacc
+	WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+	DEPENDS syntax/syntax.yacc
+	COMMENT "Generate YACC - y.tab.c/.h")
+
+ADD_CUSTOM_COMMAND(
+	OUTPUT ${CMAKE_BINARY_DIR}/lex.yy.c
+	COMMAND lex -d ${CMAKE_SOURCE_DIR}/syntax/syntax.lex
```

Comment:
According to the original gen_syntax.sh script, the lex command has no -d option.
Please refer followings:

```
#!/bin/bash
yacc -d syntax.yacc
lex syntax.lex
mv lex.yy.c ../src
mv y.tab.* ../src
```

Reply:
Oops! sorry :(
I will update the commit ASAP. Thank you for review !